src/sys/unix.rs

```diff
@@ -1773,6 +1803,41 @@ impl crate::Socket {
                 })
         }
     }
+
+    /// Attach Berkeley Packet Filter(BPF) on this socket.
+    ///
+    /// BPF allows a user-space program to attach a filter onto any socket
+    /// and allow or disallow certain types of data to come through the socket.
+    ///
+    /// For more information about this option, see [filter](https://www.kernel.org/doc/html/v5.12/networking/filter.html)
+    #[cfg(all(feature = "all", target_os = "linux"))]
```

Comment:
Does Android support this?

Reply:
According to the filter [documentation](https://android.googlesource.com/kernel/msm/+/refs/tags/android-4.4w_r10/Documentation/networking/filter.txt) in android kernel, it should be supported.

I will fix it.